### PR TITLE
Create error wrapper and custom error

### PR DESCRIPTION
**Pull request by Emanalabsi**
Relates to #7

### CODE REVIEW

File: server/utils/errorHandler/errorWrapper.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Request, Response, NextFunction, RequestHandler} from "express";
+
+const errorMappings: { [name: string]: number } ={
+  JsonWebTokenError: 401,
+  Unauthorized: 401,
+  ValidationError: 422,
+  "NotFound": 404
+};
+
+const errorWrapper = async(controller: RequestHandler,req:Request, res:Response, next:NextFunction) =>{
```
Comment (Saeed99Madi) [inline server/utils/errorHandler/errorWrapper.ts:10]: Error Wrapper, needs to take just one Parameter, errorWrapper(controller: RequestHandler),
To use It in Routes Files like this : 
`
router.get('/my-route', errorWrapper(myController));
`

Comment (Emanalabsi) [reply server/utils/errorHandler/errorWrapper.ts:10]: done, will be solved in seconds

File: server/utils/errorHandler/errorWrapper.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  try {
+    await controller(req,res,next)
+  }
+  catch (error:any) {
```
Comment (Saeed99Madi) [inline server/utils/errorHandler/errorWrapper.ts:14]: Using the unknown type in TypeScript is generally a better alternative than using the any type when you need to represent values of an unknown type.

**Review by alaaahmade — APPROVED** [review]

**Review by Saeed99Madi — APPROVED** [review]
Well Done @Emanalabsi

**Review by Heba-Adwan — APPROVED** [review]

**Review by ahmedirheem — APPROVED** [review]

**Pull request merged**